File: packages/python/src/unmsm_sum_api/schemas/student.py
from typing import Any

from .common import BaseResponse, FlexibleModel


class StudentInfo(FlexibleModel):
    apePaterno: str
    apeMaterno: str
    nombre: str
    codFacultad: int
    desFacultad: str
    foto: str
    sexo: str
    correoInstitucional: str
    dni: str
    codAlumno: str
    codEscuela: int
    desEscuela: str
    codEspecialidad: int
    desEspecialidad: str
    codPlan: str
    desPlan: str
    ponderado: float
    actualizoFormulario: bool
    periodo: str
    urlFoto: str
    codPermanencia: str
    desPermanencia: str
    codSituacion: str
    desSituacion: str
    regimen: str
    egresadoEG: str
    anioIngreso: int
    matricula: Any | None


class PersonalInformationResponse(BaseResponse):
    data: dict[str, Any]

File: packages/js/src/schemas/student.ts
import { z } from "zod";

import { BaseResponseSchema } from "./common.js";

export const StudentInfoSchema = z.object({
  apePaterno: z.string(),
  apeMaterno: z.string(),
  nombre: z.string(),
  codFacultad: z.number(),
  desFacultad: z.string(),
  foto: z.string(),
  sexo: z.string(),
  correoInstitucional: z.string(),
  dni: z.string(),
  codAlumno: z.string(),
  codEscuela: z.number(),
  desEscuela: z.string(),
  codEspecialidad: z.number(),
  desEspecialidad: z.string(),
  codPlan: z.string(),
  desPlan: z.string(),
  ponderado: z.number(),
  actualizoFormulario: z.boolean(),
  periodo: z.string(),
  urlFoto: z.string(),
  codPermanencia: z.string(),
  desPermanencia: z.string(),
  codSituacion: z.string(),
  desSituacion: z.string(),
  regimen: z.string(),
  egresadoEG: z.string(),
  anioIngreso: z.number(),
  matricula: z.unknown().nullable(),
}).passthrough();

export const PersonalInformationResponseSchema = BaseResponseSchema.extend({
  data: z.object({}).passthrough(),
}).passthrough();

export type StudentInfo = z.infer<typeof StudentInfoSchema>;
export type PersonalInformationResponse = z.infer<typeof PersonalInformationResponseSchema>;
